### ISSUE TITLE
Intel gadget fail

### ISSUE THREAD

**Issue by rmsa5** [qnxor/macoh#12]
Hi,
Whenever running macoh.sh script I get the following error :

> Fetching and installing Intel Power Gadget into /Applications ...
> ######################################################################## 100.0%
> error:  cannot create /Users/sashimi/macoh/tmp/Intel-� Power Gadget v3.0.3.dmg
>         Illegal byte 

I've manually installed Intel Power gadget, which by the way is now in v3.5.1 version.

Is there no way to run macoh and make it use some available version of intel power gadget instead of wanting to installing it (old) version and failing to do so ?

**Comment by sashimiman** [comment]
Yeah, well i was looking for a stress tool for my mac (it seems to reach 98°C !) and as geekbench 4 removed all stress features, I was advised to try you tool :) 

After macoh.conf editing accordingly :
```
Fetching and installing Intel Power Gadget into /Applications ...
######################################################################## 100.0%
[/Users/sashimi/macoh/tmp/ipg.zip]
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /Users/sashimi/macoh/tmp/ipg.zip or
        /Users/sashimi/macoh/tmp/ipg.zip.zip, and cannot find /Users/sashimi/macoh/tmp/ipg.zip.ZIP, period.
```
it would actually be much easier if macoh could simply use igp (and cry if not installed) rather than going through the hassle of installing it itself...)

**Comment by qnxor** [comment]
I see Intel switched from a zip file to a dmg file, but macoh still tries to unzip it. That's easy to fix.

> it would actually be much easier if macoh could simply use igp (and cry if not installed) rather than going through the hassle of installing it itself...)

It actually should, that's how I coded it if I recall correctly. It seems it can't detect this new version when it's already installed.

Both of the above problems should be easy to fix. If you have any experience with bash then feel free to look inside macoh.sh and have a go.

**Comment by yasha282** [comment]
Hi, I believe the current ipg version uses a slighty different format with the results .csv.
my approach was to run macoh.sh over and over again and fix the lines where it apparently failed. 
the first run downloaded intel power gadget 3.0.3, then it failed to unpack it. I manually installed that ipg.zip from the tmp folder and commented several lines out in macoh.sh that tried to unpack and install it. (im new to github. should i paste the code here or is there any other, better space to do that?)

the next issue i encountered was with the graphics layout engine. the download link was not correct anymore. i downloaded that manually and renamed it to what macoh expected it in tmp (gle.dmg i believe). i let macoh.sh install it from there just by running it again. the rest worked fine if i remember correctly. gputest and the encoding work beautifully. the prime torture test and fft both do not start however.

**Comment by Faddiie** [comment]
Comment this line in macoh.sh	
#unzip -q -o $tmp/ipg.zip -d $tmp
and change these lines as well

**Comment by Faddiie** [comment]
Comment this line in macoh.sh
#unzip -q -o $tmp/ipg.zip -d $tmp
and change these lines as well	
wget $tmp/ipg.dmg "$url_ipg" -#
mnt $tmp/ipg*.dmg

and add change the url to
url_ipg="https://software.intel.com/file/613985/download"

it ill work
